synapse/lib/ast.py

```diff
@@ -347,52 +349,68 @@ def getRuntVars(self, runt):
 
     async def run(self, runt, genr):
 
-        kvar = self.kids[0].value()
-        ivar = self.kids[1].value()
         subq = self.kids[2]
+        name = self.kids[0].value()
 
-        items = runt.getVar(ivar)
-        if items is None:
-            raise s_exc.NoSuchVar(name=ivar)
-
-        broke = False
-        for item in items:
+        count = 0
+        async for node, path in genr:
 
-            if isinstance(self.kids[0], VarList):
+            count += 1
 
-                if len(item) != len(kvar):
-                    raise s_exc.StormVarListError(names=kvar, vals=item)
+            for item in await self.kids[1].compute(path):
 
-                for name, valu in zip(kvar, item):
-                    runt.vars[name] = valu
+                if isinstance(name, (list, tuple)):
 
-            else:
-                runt.vars[kvar] = item
+                    if len(name) != len(item):
+                        raise s_exc.StormVarListError(names=name, vals=item)
 
-            newg = subq.inline(runt, genr)
+                    for x, y in itertools.zip_longest(name, item):
+                        path.set(x, y)
+                        runt.setVar(x, y)
 
-            while True:
+                else:
+                    # set both so inner subqueries have it in their runtime
+                    path.set(name, item)
+                    runt.setVar(name, item)
 
                 try:
 
-                    async for node, path in newg:
-                        yield node, path
+                    newg = agen((node, path))
+                    await s_common.aspin(subq.inline(runt, newg))
 
+                except StormBreak:
                     break
 
-                except StormBreak as e:
-                    broke = True
-                    if e.item is not None:
-                        yield e.item
-                    break
+                except StormContinue:
+                    continue
+
+            yield node, path
+
+        # no nodes and a runt safe value should execut once
```

Comment:
execut->execute

Reply:
uh... *shifty eyes* i totally just left that to um... make sure you guys were... paying attention. :D